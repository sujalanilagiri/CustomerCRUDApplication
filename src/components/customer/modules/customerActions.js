export const GET_CUSTOMERS_LIST = "GET_CUSTOMER_LIST";

export const GET_CUSTOMERS = "GET_CUSTOMERS";

export const SAVE_CUSTOMER = "SAVE_CUSTOMER";

export const DELETE_CUSTOMER = "DELETE_CUSTOMER";

export const UPDATE_CUSTOMER = "UPDATE_CUSTOMER";

export const UPDATE_CUSTOMER_RESULT = "UPDATE_CUSTOMER_RESULT";

export const SAVE_CUSTOMER_RESULT = "SAVE_CUSTOMER_RESULT";

export function getCustomerList() {
  return {
    type: GET_CUSTOMERS
  };
}

export function saveCustomer(data) {
  return {
    type: SAVE_CUSTOMER,
    data:data
  };
}

export function deleteCustomer(name) {
  return {
    type: DELETE_CUSTOMER,
    customerName:name
  };
}

export function updateCustomer(customer) {
  return {
    type: UPDATE_CUSTOMER,
    updateCustomer:customer
  };
}