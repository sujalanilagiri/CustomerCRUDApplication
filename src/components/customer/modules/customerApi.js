import axios from "axios";

const httpClient = axios.create();
httpClient.defaults.timeout = 150000;

export default class CustomerApi {
  static get() {
    return axios
      .get("http://127.0.0.1:5000/customers/all")
      .catch(function(error) {
        console.log(error);
      });
  }
  static post(body) {
    console.log('in sagas',body);
    const firstName = body.firstName ;
    return axios.post(`http://127.0.0.1:5000/customer/${firstName}`, {
      lastName:body.lastName,
      age:body.age,
      email:body.email
    });
  }

  static put(body) {
    console.log('in sagas',body);
    const firstName = body.firstName ;
    return axios.put(`http://127.0.0.1:5000/customer/${firstName}`, {
      lastName:body.lastName,
      age:body.age,
      email:body.email
    });
  }

  static delete(name) {
    console.log('in sagas',name);
    const firstName = name ;
    return axios.delete(`http://127.0.0.1:5000/customer/${firstName}`);
  }
}