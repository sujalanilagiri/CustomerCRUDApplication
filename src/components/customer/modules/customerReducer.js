import {
  GET_CUSTOMERS,
  GET_CUSTOMERS_LIST,
  SAVE_CUSTOMER,
  DELETE_CUSTOMER,
  SAVE_CUSTOMER_RESULT,
  UPDATE_CUSTOMER,
  UPDATE_CUSTOMER_RESULT
} from "./customerActions";

const customerReducer = (state = {}, action) => {
  switch (action.type) {
    case GET_CUSTOMERS:
    case GET_CUSTOMERS_LIST:
      return {
        ...state,
        customerList: action.data
      };
    case SAVE_CUSTOMER:
      return {
        ...state,
        savedCustomer: action.data
      };
    case DELETE_CUSTOMER:
      return {
        ...state,
        selectedCustomer: action.customerName
      };
    case UPDATE_CUSTOMER:
      return {
        ...state,
        updatedCustomer: action.updateCustomer
      };

    case SAVE_CUSTOMER_RESULT:
      return {
        ...state,
        response: action.val,
    
      };

    case UPDATE_CUSTOMER_RESULT:
          return {
            ...state,
            response: action.val,
        
          };
    default:
      return {
        ...state,
        customerList: action.data
      };
  }
};

export default customerReducer;
