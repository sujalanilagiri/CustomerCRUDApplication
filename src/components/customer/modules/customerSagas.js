
import {takeEvery,call,put,select} from "redux-saga/effects";
import CustomerApi from "./customerApi";

export const getSavedCustomer = state => state.customers.savedCustomer;

export const getSelectedCustomer = state => state.customers.selectedCustomer;

export const getUpdatedCustomer = state => state.customers.updatedCustomer;
function* fetchData(action) {
  try {
    const data = yield call(CustomerApi.get);
    if (data) {
      yield put({ type: "GET_CUSTOMERS_LIST", data });
    }
  } catch (e) {
    console.log("Error" + e);
  }
}

function* postData(action) {
  try {
    const ch = yield select(getSavedCustomer);
    const val = yield call(CustomerApi.post, ch);
    if (val) {
      yield put({ type: "SAVE_CUSTOMER_RESULT", val });
      yield put({ type: 'GET_CUSTOMERS' })
    }
  } catch (e) {
    console.log("Error" + e);
  }
}

function* deleteCustomer(action) {
  try {
    const customerToDelete = yield select(getSelectedCustomer);
    const val = yield call(CustomerApi.delete, customerToDelete);
    if (val) {
      yield put({ type: "DELETE_CUSTOMER_RESULT", val });
      yield put({ type: 'GET_CUSTOMERS' })
    }
  } catch (e) {
    console.log("Error" + e);
  }
}
function* updateCustomer(action) {
  try {
    const updatedCustomer = yield select(getUpdatedCustomer);
    const val = yield call(CustomerApi.put, updatedCustomer);
    if (val) {
      yield put({ type: "UPDATE_CUSTOMER_RESULT", val });
      yield put({ type: 'GET_CUSTOMERS' })
    }
  } catch (e) {
    console.log("Error" + e);
  }
}

function* watchsaga() {
    yield takeEvery("GET_CUSTOMERS", fetchData);
    yield takeEvery("SAVE_CUSTOMER", postData);
    yield takeEvery("DELETE_CUSTOMER",deleteCustomer);
    yield takeEvery("UPDATE_CUSTOMER",updateCustomer);
  }
  export const customerSagas = [watchsaga()];