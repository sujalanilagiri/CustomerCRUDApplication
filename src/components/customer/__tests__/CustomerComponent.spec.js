import {configure} from 'enzyme';
import configureStore from "redux-mock-store";
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import {shallow} from 'enzyme';
import {getCustomerList,saveCustomer,deleteCustomer,updateCustomer} from "../modules/customerActions";
import customerReducer from "../modules/customerReducer";


configure({adapter:new Adapter()});

const users = [
    {
        "firstName": "Nicholas",
        "lastName" : "Thomas",
        "age": 42,
        "email": "nicholas.thomas@gmail.com"
    },
    {
        "firstName": "Elvin",
        "lastName" :"Martin",
        "age": 32,
        "email": "martin.elvin@gmail.com"
    },
    {
        "firstName": "Jass",
        "lastName" :"Base",
        "age": 22,
        "email": "jass.base@gmail.com"
    }
];
describe("ACTIONS: customerActions",()=>{
    const mockStore = configureStore();
    const store = mockStore();
    beforeEach(()=>{
        store.clearActions();
    });

    test("ACTION: GET_CUSTOMERS",()=>{
        store.dispatch(getCustomerList());
        expect(store.getActions()).toMatchSnapshot();
    })
    test("ACTION: SAVE_CUSTOMER",()=>{
        store.dispatch(saveCustomer());
        expect(store.getActions()).toMatchSnapshot();
    })
    test("ACTION: DELETE_CUSTOMER",()=>{
        store.dispatch(deleteCustomer());
        expect(store.getActions()).toMatchSnapshot();
    })
    test("ACTION: UPDATE_CUSTOMERS",()=>{
        store.dispatch(updateCustomer());
        expect(store.getActions()).toMatchSnapshot();
    })
});

describe("REDUCER: customerReducer",()=>{
    test("GET_CUSTOMERS",()=>{
        expect(customerReducer(undefined,getCustomerList(users))).toMatchSnapshot();
    })
})



