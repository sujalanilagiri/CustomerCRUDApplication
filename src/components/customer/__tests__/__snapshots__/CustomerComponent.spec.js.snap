// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ACTIONS: customerActions ACTION: DELETE_CUSTOMER 1`] = `
Array [
  Object {
    "customerName": undefined,
    "type": "DELETE_CUSTOMER",
  },
]
`;

exports[`ACTIONS: customerActions ACTION: GET_CUSTOMERS 1`] = `
Array [
  Object {
    "type": "GET_CUSTOMERS",
  },
]
`;

exports[`ACTIONS: customerActions ACTION: SAVE_CUSTOMER 1`] = `
Array [
  Object {
    "data": undefined,
    "type": "SAVE_CUSTOMER",
  },
]
`;

exports[`ACTIONS: customerActions ACTION: UPDATE_CUSTOMERS 1`] = `
Array [
  Object {
    "type": "UPDATE_CUSTOMER",
    "updateCustomer": undefined,
  },
]
`;

exports[`REDUCER: customerReducer GET_CUSTOMERS 1`] = `
Object {
  "customerList": undefined,
}
`;
